airflow/utils/python_virtualenv.py

```diff
@@ -87,14 +87,15 @@ def prepare_virtualenv(
     :param system_site_packages: Whether to include system_site_packages in your virtualenv.
         See virtualenv documentation for more information.
     :type system_site_packages: bool
-    :param requirements: List of additional python packages
-    :type requirements: List[str]
+    :param requirements: Path to the requirements.txt file
```

Comment:
I think we should still (backwards compatibility) handle the case where requirements are List[str]. It will make it a bit more complex, but I think it is needed. 

I propose to keep the old `requrements` handling only List[str] and add a new parameter `requirements_file_path`  - and check if only one of those is passed (and act accordingly). I think it is very ambiguous to name `requirements` something that is path to requirements file.

Reply:
Addressed! 😃